src/main/java/edu/virginia/lib/indexing/ASpaceObject.java

```diff
@@ -515,4 +523,100 @@ public void printOutRawData() {
         JsonHelper.writeOutJson(record);
     }
 
+    public void writeCirculationRecord(File file) throws IOException {
+    	//make MARC record with 245 and 590 fields
+		MarcFactory factory = MarcFactory.newInstance();
+		Record r = factory.newRecord();
+		DataField df;
+		Subfield sf;
+		
+		String title = record.getString("title");
+        char nonIndexChars = '0';
+        if (title.startsWith("A "))
+        	nonIndexChars = '2';
+        else if (title.startsWith("The "))
+        	nonIndexChars = '4';
+        
+        df = factory.newDataField("245", '0', nonIndexChars);
+		sf = factory.newSubfield('a', title);
+		df.addSubfield(sf);
+		r.addVariableField(df);
+		
+		df = factory.newDataField("590", '1', ' ');
+		sf = factory.newSubfield('a', "From ArchivesSpace: " + record.getString("record_uri"));
+		df.addSubfield(sf);
+		r.addVariableField(df);
+		
+		
+		
+		//pull desired resource from ArchivesSpace, get reference for each unique top_container
+        HashSet<String> topContainers = new HashSet<String>();
+        JsonArray children = record.getJsonArray("children");
+        
+        for (JsonValue child : children) {
```

Comment:
This will likely have to be a separate method that is called recursively because children can have children and so forth.